typescript/infra/scripts/verify-validator.ts

```diff
@@ -104,170 +89,259 @@ class S3Wrapper {
     const diffS =
       (prospectiveLastMod.valueOf() - controlLastMod!.valueOf()) / 1000;
     if (Math.abs(diffS) > 10) {
-      console.log(`${i}: Modification times differ by ${diffS}s`);
+      console.log(`${checkpointIndex}: Modification times differ by ${diffS}s`);
     }
-    modTimeDeltasS.push(diffS);
-    fullyCorrectCheckpoints.push(i);
+    this.modTimeDeltasS.push(diffS);
+    this.fullyCorrectCheckpoints.push(checkpointIndex);
   }
 
-  console.log(
-    `Fully correct checkpoints (${fullyCorrectCheckpoints.length}): ${fullyCorrectCheckpoints}\n`,
-  );
-  if (extraCheckpoints.length)
-    console.log(
-      `Extra checkpoints (${extraCheckpoints.length}): ${extraCheckpoints}\n`,
-    );
-  if (missingCheckpoints.length)
-    console.log(
-      `Missing checkpoints (${missingCheckpoints.length}): ${missingCheckpoints}\n`,
-    );
-  if (invalidCheckpoints.length)
-    console.log(
-      `Invalid checkpoints (${invalidCheckpoints.length}): ${invalidCheckpoints}\n`,
-    );
+  private async getLatestCheckpoints(): Promise<{
+    controlLatestCheckpoint: number;
+    prospectiveLastCheckpoint: number;
+  }> {
+    const [
+      { obj: controlLatestCheckpoint },
+      { obj: prospectiveLastCheckpoint },
+    ] = await Promise.all([
+      this.controlS3BucketClient
+        .getS3Obj<number>('checkpoint_latest_index.json')
+        .catch((err) => {
+          console.error(
+            "Failed to get control validator's latest checkpoint.",
+            err,
+          );
+          process.exit(1);
+        }),
+      this.prospectiveS3BucketClient
+        .getS3Obj<number>('checkpoint_latest_index.json')
+        .catch((err) => {
+          console.error(
+            "Failed to get prospective validator's latest checkpoint.",
+            err,
+          );
+          process.exit(1);
+        }),
+    ]);
 
-  if (modTimeDeltasS.length > 1) {
-    // Drop the time of the first one since it is probably way off
-    modTimeDeltasS.length--;
-    console.log(
-      `Time deltas (∆ < 0 -> prospective came earlier than the control)`,
-    );
-    console.log(modTimeDeltasS);
-    console.log(`Median: ${median(modTimeDeltasS)}s`);
-    console.log(`Mean:   ${mean(modTimeDeltasS)}s`);
-    console.log(`Stdev:  ${stdDev(modTimeDeltasS)}s`);
+    if (
+      !isLatestCheckpoint(controlLatestCheckpoint) ||
+      !isLatestCheckpoint(prospectiveLastCheckpoint)
+    )
+      throw new Error('Invalid latest checkpoint data');
+
+    console.log(`Latest Index`);
+    console.log(`control: ${controlLatestCheckpoint}`);
+    console.log(`prospective: ${prospectiveLastCheckpoint}\n`);
+    return { controlLatestCheckpoint, prospectiveLastCheckpoint };
+  }
+
+  private async getControlCheckpoint(
+    checkpointIndex: number,
+  ): Promise<{ control: Checkpoint; controlLastMod: Date } | null> {
+    let control: Checkpoint, controlLastMod: Date, unrecoverableError;
+    try {
+      const t = await this.controlS3BucketClient.getS3Obj(
```

Comment:
Even if it is a temporary variable, I think unless its a one-line lambda, I would probably suggest to give it a more descriptive name
```suggestion
      const s3Object = await this.controlS3BucketClient.getS3Obj(
```